src/System.Linq.Parallel/tests/QueryOperators/MaxTests.cs

```diff
@@ -0,0 +1,416 @@
+            Assert.Equal(cs.Token, oe.CancellationToken);
+            oe = Assert.Throws<OperationCanceledException>(() => ParallelEnumerable.Repeat((long?)0, 2).WithCancellation(cs.Token).Max());
+            Assert.Equal(cs.Token, oe.CancellationToken);
+            oe = Assert.Throws<OperationCanceledException>(() => ParallelEnumerable.Repeat((long?)0, 2).WithCancellation(cs.Token).Max(x => x));
+            Assert.Equal(cs.Token, oe.CancellationToken);
+
+            oe = Assert.Throws<OperationCanceledException>(() => ParallelEnumerable.Repeat((float)0, 2).WithCancellation(cs.Token).Max());
+            Assert.Equal(cs.Token, oe.CancellationToken);
+            oe = Assert.Throws<OperationCanceledException>(() => ParallelEnumerable.Repeat((float)0, 2).WithCancellation(cs.Token).Max(x => x));
+            Assert.Equal(cs.Token, oe.CancellationToken);
+            oe = Assert.Throws<OperationCanceledException>(() => ParallelEnumerable.Repeat((float?)0, 2).WithCancellation(cs.Token).Max());
+            Assert.Equal(cs.Token, oe.CancellationToken);
+            oe = Assert.Throws<OperationCanceledException>(() => ParallelEnumerable.Repeat((float?)0, 2).WithCancellation(cs.Token).Max(x => x));
+            Assert.Equal(cs.Token, oe.CancellationToken);
+
+            oe = Assert.Throws<OperationCanceledException>(() => ParallelEnumerable.Repeat((double)0, 2).WithCancellation(cs.Token).Max());
+            Assert.Equal(cs.Token, oe.CancellationToken);
+            oe = Assert.Throws<OperationCanceledException>(() => ParallelEnumerable.Repeat((double)0, 2).WithCancellation(cs.Token).Max(x => x));
+            Assert.Equal(cs.Token, oe.CancellationToken);
+            oe = Assert.Throws<OperationCanceledException>(() => ParallelEnumerable.Repeat((double?)0, 2).WithCancellation(cs.Token).Max());
+            Assert.Equal(cs.Token, oe.CancellationToken);
+            oe = Assert.Throws<OperationCanceledException>(() => ParallelEnumerable.Repeat((double?)0, 2).WithCancellation(cs.Token).Max(x => x));
+            Assert.Equal(cs.Token, oe.CancellationToken);
+
+            oe = Assert.Throws<OperationCanceledException>(() => ParallelEnumerable.Repeat((decimal)0, 2).WithCancellation(cs.Token).Max());
+            Assert.Equal(cs.Token, oe.CancellationToken);
+            oe = Assert.Throws<OperationCanceledException>(() => ParallelEnumerable.Repeat((decimal)0, 2).WithCancellation(cs.Token).Max(x => x));
+            Assert.Equal(cs.Token, oe.CancellationToken);
+            oe = Assert.Throws<OperationCanceledException>(() => ParallelEnumerable.Repeat((decimal?)0, 2).WithCancellation(cs.Token).Max());
+            Assert.Equal(cs.Token, oe.CancellationToken);
+            oe = Assert.Throws<OperationCanceledException>(() => ParallelEnumerable.Repeat((decimal?)0, 2).WithCancellation(cs.Token).Max(x => x));
+            Assert.Equal(cs.Token, oe.CancellationToken);
```

Comment:
This pattern occurs a lot throughout these tests.  What about adding a helper like:
```C#
internal static void AssertOperationCanceledException(Action action, CancellationToken cancellationToken)
{
    var oce = Assert.Throws<OperationCanceledException>(action);
    Assert.Equals(cancellationToken, oce.CancellationToken);
}
```
such that all of these two-line call sites come down to a single line?